src/main/java/core/basesyntax/services/handlers/DecreaseFruitHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.services.handlers;
+
+import core.basesyntax.model.Operation;
+import java.util.Map;
+
+public class DecreaseFruitHandler implements FruitHandler {
+    @Override
+    public int newQuantity(Operation operation, Map<String, Integer> fruitStorage) {
+        for (Map.Entry<String, Integer> entry : fruitStorage.entrySet()) {
+            if (entry.getKey().equals(operation.getFruit())) {
+                return entry.getValue() - operation.getQuantity();
+            }
+        }
+        return 0;
```

Comment:
perhaps it is worth to throw some exception here(`0` may be a valid value). We can carry out operations only on fruits that exist in Storage, if there is no such fruit, it is better to throw an exception and immediately report that something went wrong.